samples/cellular/modem_shell/src/sms/sms_shell.c

```diff
@@ -161,34 +136,85 @@ static int sms_shell(const struct shell *shell, size_t argc, char **argv)
 		}
 	}
 
-	/* Run given command with it's arguments */
-	switch (command) {
-	case SMS_CMD_REGISTER:
-		err = sms_register();
-		break;
-	case SMS_CMD_UNREGISTER:
-		err = sms_unregister();
-		break;
-	case SMS_CMD_SEND:
-		err = sms_send_msg(arg_number, arg_message, arg_message_type);
-		break;
-	case SMS_CMD_RECV:
-		err = sms_recv(arg_receive_start);
-		break;
-	default:
-		mosh_error("Internal error. Unknown sms command=%d", command);
-		err = -EINVAL;
-		break;
+	err = sms_send_msg(arg_number, arg_message, arg_message_type);
+
+	return err;
+
+show_usage:
+	/* Reset getopt for another users */
+	optreset = 1;
+
+	sms_print_usage(SMS_CMD_SEND);
+	return err;
+}
+
+static int cmd_sms_recv(const struct shell *shell, size_t argc, char **argv)
+{
+	int err = 0;
+
+	/* Variables for command line arguments */
+	bool arg_receive_start = false;
+
+	optreset = 1;
+	optind = 1;
+	int opt;
+
+	while ((opt = getopt_long(argc, argv, "rh", long_options, NULL)) != -1) {
+
+		switch (opt) {
+		case 'r': /* Start monitoring received messages */
+			arg_receive_start = true;
+			break;
+		case '?':
+			mosh_error("Unknown option. See usage:");
+			goto show_usage;
+		case 'h':
+		default:
+			goto show_usage;
+		}
 	}
 
+	err = sms_recv(arg_receive_start);
+
 	return err;
 
 show_usage:
 	/* Reset getopt for another users */
 	optreset = 1;
```

Comment:
I'm removing `optreset = 1` from the end of the functions because it's not everywhere and it's in the beginning for all functions.